src/gravitational_waves/gw.cpp

```diff
@@ -15,28 +15,55 @@ namespace BSMPT
 GravitationalWave::GravitationalWave(BounceSolution &BACalc,
                                      const int &which_transition_temp)
 {
-  data.transitionTemp = BACalc.CalcTransitionTemp(which_transition_temp);
+  BACalc.SetAndCalculateGWParameters(which_transition_temp);
+  data.transitionTemp = BACalc.GetTransitionTemp();
+  data.reheatingTemp  = BACalc.GetReheatingTemp();
   data.PTStrength     = BACalc.GetPTStrength();
-  data.InvTimeScale   = BACalc.GetInvTimeScale();
-  data.vb             = BACalc.GetWallVelocity();
-  data.kappa_sw       = Getkappa_sw(
-      data.PTStrength, data.vb, BACalc.modelPointer->SMConstants.Csound);
-  data.K_sw = GetK_sw(
-      data.PTStrength, data.vb, BACalc.modelPointer->SMConstants.Csound);
-  data.HR = GetHR(
-      data.InvTimeScale, data.vb, BACalc.modelPointer->SMConstants.Csound);
-  data.kappa_turb = CalcEpsTurb(BACalc.GetEpsTurb()) * data.kappa_sw;
-  data.K_turb     = GetK_turb(data.PTStrength, data.kappa_turb);
-  data.gstar      = BACalc.GetGstar();
-  data.Hstar      = BACalc.HubbleRate(data.transitionTemp);
-
-  if (data.InvTimeScale < 1)
+  data.betaH          = BACalc.GetInvTimeScale();
+  data.vw             = BACalc.GetWallVelocity();
+
+  // Sound speeds
+  data.Csound_false = BACalc.GetSoundSpeedFalse();
+  data.Csound_true  = BACalc.GetSoundSpeedTrue();
+
+  // General purpose GW parameters
+  data.HR    = BACalc.GetRstar() * BACalc.HubbleRate(data.transitionTemp);
+  data.gstar = BACalc.GetGstar(data.transitionTemp);
+  data.FGW0  = 1.64 / pow(h, 2) * 1.e-5 *
+              pow(100. / BACalc.GetGstar(data.transitionTemp), 1 / 3.);
+  data.Hstar0 =
+      GetHstar0(data.reheatingTemp, BACalc.GetGstar(data.transitionTemp));
+  // Collisions
+  data.pnlo_scaling = BACalc.pnlo_scaling;
+  data.kappa_col    = kappa::Getkappa_col(
+      data.transitionTemp, data.pnlo_scaling, data.HR, BACalc);
+  // Turbulence
+  data.Epsilon_Turb = BACalc.GetEpsTurb();
+  // Sound waves
+  const double alpha_eff =
+      (1 - data.kappa_col) *
+      data.PTStrength; // remove energy that goes into collisions
+  data.vCJ = BACalc.GetChapmanJougetVelocity();
+  std::vector<std::vector<double>> vprofile; // Fluid profile
+  data.kappa_sw = (alpha_eff / data.PTStrength) *
+                  kappa::kappaNuMuModel(pow(data.Csound_true, 2),
+                                        pow(data.Csound_false, 2),
+                                        alpha_eff,
+                                        BACalc.vwall,
+                                        vprofile);
+  // XiShock = fastest fluid shell
+  if (data.kappa_sw > 0)
+    data.XiShock = vprofile.back().front();
+  else
+    data.XiShock = 1.; // Does not matter as efficiency if zero
```

Comment:
```suggestion
  if (data.kappa_sw > 0)
    {
      data.XiShock = vprofile.back().front();
    }
  else
    {
      data.XiShock = 1.; // Does not matter as efficiency if zero
     }
```

Just to match the style of the rest of the code